### DISCUSSION TITLE
How to use UnifiedJedis with JedisSentinelPool

### DISCUSSION THREAD

**Discussion by Datta19** [xetorthio/jedis#3526] [Q&A] [answered]
we are using jedis version 4.4.0. we are using UnifiedJedis client in our java application , for high availablity we need JedisSentinelPool so how can we use JedisSentinelPool with UnifiedJedis.

**Comment by sazzad16** [comment]
Check JedisSentineled.

**Reply by Datta19** [reply]
@sazzad16 Thanks for the quick reply,  could you please give us sample code of it or any sample code link, so we can refer it for configure the UnifiedJedis with Sentinel.

**Reply by sazzad16** [reply]
@Datta19 No, sorry.
But configuration of JedisSentineled and JedisSentinelPool are "similar".

**Reply by sazzad16** [reply]
@Datta19 If you provide your JedisSentinelPool configuration, we may give it a try for JedisSentineled provided it won't be a huge effort. PLEASE TRY YOURSELF FIRST.

**Reply by hemangi225** [reply]
Hi @sazzad16,

Can you please provide the details if we can use JedisSentinelPool in UnifiedJedis or not?

We have tried with JedisSentineled but there is no any option with JedisSentineled  to provide **executor** e.g. 
 public UnifiedJedis(ConnectionProvider provider, int maxAttempts, Duration maxTotalRetriesDuration) {
    this.provider = provider;
    **this.executor = new RetryableCommandExecutor(provider, maxAttempts, maxTotalRetriesDuration);**
    this.commandObjects = new CommandObjects();
    this.graphCommandObjects = new GraphCommandObjects(this);
    this.graphCommandObjects.setBaseCommandArgumentsCreator((comm) -> this.commandObjects.commandArguments(comm));
  }


Can you please give us the detail that how it can be implemented using JedisSentineled or using JedisSentinelPool ?

For your reference,

Below is our code block without sentinel and we want to introduce sentinel.

final int maxAttempts = IMMConstants.redisMaxAttempts;
final Duration maxRetriesDuration = Duration.ofSeconds(IMMConstants.redisMaxRetriesDuration);

PooledConnectionProvider pooledConnectionProvider = new PooledConnectionProvider(config);
UnifiedJedis jedisClient = new UnifiedJedis(pooledConnectionProvider, maxAttempts, maxRetriesDuration);

**Reply by sazzad16** [reply, marked as answer]
@hemangi225 

SentineledConnectionProvider sentineledConnectionProvider = new SentineledConnectionProvider(masterName, masterConfig, sentinels, sentinelConfig);
UnifiedJedis jedisClient = new UnifiedJedis(sentineledConnectionProvider, maxAttempts, maxRetriesDuration);

**Reply by Datta19** [reply]
@sazzad16  Thanks for the detailed reply.

**Reply by hemangi225** [reply]
@sazzad16 Thanks for reply.
